lib/rules/require-blocks-on-newline.js

```diff
@@ -37,27 +65,161 @@
  * if (true) { doSomething(); }
  * var abc = function() {};
  * ```
+ * ```js
+ * if (true) { //comments
+ *     doSomething();
+ *     doSomethingElse();
+ * }
+ * if (true) { doSomething(); }
+ * var abc = function() {};
+ * ```
+ *
+ * ```js
+ * if (true) {
+ *     doSomething();
+ *     doSomethingElse();
+ *     //comments }
+ * if (true) { doSomething(); }
+ * var abc = function() {};
+ * ```
  *
  * ##### Invalid
  *
  * ```js
  * if (true) { doSomething(); doSomethingElse(); }
  * ```
+ *
+ * ##### Valid for mode {
+ *      includeComments: true
+ * }
+ *
+ * ```
+ * if (true) {
+ *     //comments
+ *     doSomething();
+ * }
+ * var abc = function() {};
+ * ```
+  * ```
+ * if (true) {
+ *     doSomething();
+ *      //comments
+ * }
+ * var abc = function() {};
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```
+ * if (true) { //comments
+ *     doSomething();
+ * }
+ * var abc = function() {};
+ * ```
+  * ```
+ * if (true) {
+ *     doSomething();
+ * //comments}
+ * var abc = function() {};
+ * ```
+ *
+ * ##### Valid for mode {
+ *      includeComments: true,
+ *      minStatements: `1`
+ * }
+ *
+ * ```js
+ * if (true) {
+ *     //comments
+ *     doSomething();
```

Comment:
Well, those are not statements, they are function calls, which makes them an expressions, maybe we need to choose more general term?